### PR TITLE
Don't shell out to GPG on import

### PR DESCRIPTION
**Pull request by znewman01**
Fixes: #428

### Description of the changes being introduced by the pull request:

I recommend that you read this PR commit-by-commit.

This PR does the following:

1. Adds functions to `securesystemslib.gpg.constants` to replace the constants that depended on shelling out. (Now the file contains more than constants, oops. They can't go in util, though, because that would create a circular dependency.)
2. Replaces the constants with lazy calls to the underlying functions. This is a huge hack.
3. Use the function calls instead of the constants internally.

    If you don't want to make breaking changes, we should stop here.
4. Remove the GPG constants. This is technically a breaking change (but see the commit message for justification about why it's probably okay).
5. Catch timeouts, as originally proposed in #428.

### Please verify and check that the pull request fulfils the following requirements:

- [X] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [X] Tests have been added for the bug fix or new feature
- [X] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by jku** [comment]
> I'll have another look before clicking approve but I think this will work (and hopefully the constants can be removed a bit later).

oh, forgot to say: I am not making a statement about when the API breaking 4th commit should be included: I'll leave this to some Real Maintainers. Just saying this seems good to me with and without the last commits.

File: securesystemslib/gpg/constants.py
Comment (znewman01) [reply securesystemslib/gpg/constants.py:27]: I have to put a maxsize for py3.7, which is in the tox matrix. Otherwise would use the default.

I picked 3 because we should only call it three times under normal circumstances (`gpg`, `gpg2`, `$GNUPG`). That's perhaps a little bit too clever but I had a hard time justifying other numbers.

**Comment by ofek** [comment]
Nope, we'll update when it happens 👍

File: tox.ini
Comment (lukpueh) [reply tox.ini:42]: @znewman01, please remove unless necessary.

**Review by lukpueh — CHANGES_REQUESTED** [review]
We got a 👍 from DataDog. Please address inline comment, and we are good to go.

File: tox.ini
Comment (znewman01) [reply tox.ini:42]: It failed *every time* before, not occasionally.

Note that I set `GNUPG = false` in this environment, which will give an error. That's *different* from `unset GNUPG`, which will find no GPG. 

I don't believe that `py38-no-gpg`  tests the same thing. This test checks "if we shell out when *importing* `gpg.constants` we fail"; `py38-no-gpg` tests "if we have no GPG available, tests still pass."

It might be clearer to replace `false` with a script like `touch /tmp/oops_i_ran` and check that that file wasn't created after import but that felt a little too clever.

As to the question—is this necessary? I don't know; are any tests necessary? Without it, we're not really checking that the issue is fixed—I could revert the rest of the PR and the existing tests would still pass. Up to the maintainers whether you want a test for the fix or not.

**Review by lukpueh — APPROVED** [review]

**Comment by woodruffw** [comment]
Would it be possible to get a release cut with these changes?

It looks like the last release was cut the day before, and it'd be nice to be able to bump `securesystemslib` as a subdependency for things like `sigstore-python` to avoid these kinds of import side effects 🙂.

**Pull request merged**
